File: php-poo/aulas/aula14/Gafanhoto.php
<?php
require_once 'Pessoa.php';
class Gafanhoto extends Pessoa
{
    // Attributes
    private $login;
    private $totalAssistido;

    // Method Construct
    /**
     * @param $nome, $idade, $sexo, $login
     */
    public function __construct($nome, $idade, $sexo, $login)
    {
        // Parent constructors => Chamada ao construtor da SuperClasse (Classe Pessoa)
       parent::__construct($nome, $idade, $sexo);

       $this->login = $login;
       $this->totalAssistido = 0;
    }


    // Method
    public function assistirMaisUm() {
        $this->totalAssistido += 1;
    }


    // Special Methods, Getters and Setters
    public function getLogin()
    {
        return $this->login;
    }

    public function setLogin($login): void
    {
        $this->login = $login;
    }

     public function getTotalAssistido(): int
    {
        return $this->totalAssistido;
    }

    public function setTotalAssistido(int $totalAssistido): void
    {
        $this->totalAssistido = $totalAssistido;
    }

}

File: java-poo/projects/poo-aula15/src/aula15/Gafanhoto.java
package aula15;

public class Gafanhoto extends Pessoa {
    // Attributes
    private String login;
    private int totalAssistido;

    // Construcor
    public Gafanhoto(String nome, int idade, String sexo, String login) {
        super(nome, idade, sexo);
        this.login = login;
        this.totalAssistido = 0;
    }

    @Override
    public String toString() {
        return "Gafanhoto{" +
                super.toString() +
                "login='" + login + '\'' +
//                ", nome='" + nome + '\'' +
//                ", idade=" + idade +
//                ", sexo='" + sexo + '\'' +
                ", totalAssistido='" + totalAssistido + '\'' +
                '}';
    }

    // Method
    public void viuMaisUm(){
        // TODO
    }

    // Special Method - Getters and Setters
    public String getLogin() {
        return login;
    }

    public void setLogin(String login) {
        this.login = login;
    }

    public int getTotalAssistido() {
        return totalAssistido;
    }

    public void setTotalAssistido(int totalAssistido) {
        this.totalAssistido = totalAssistido;
    }
}
